enclave-runner/src/usercalls/mod.rs

```diff
@@ -229,59 +229,99 @@ struct RunningTcs {
     event_queue: Receiver<u8>,
 }
 
-pub(crate) struct EnclaveState {
+trait EnclaveKind: Send + Sync {
```

Comment:
How about `enum EnclaveKind { Command(Command), Library(Library) }` for clarity and to avoid the `Box::new` allocation?

Reply:
Sure I can do that